src/main/groovy/com/palantir/gradle/docker/DockerRunPlugin.groovy

```diff
@@ -73,11 +78,30 @@ class DockerRunPlugin implements Plugin<Project> {
 
             dockerRun.with {
                 List<String> args = Lists.newArrayList()
-                args.addAll(['docker', 'run', '-d'])
+                args.addAll(['docker', 'run'])
+                if (ext.daemonize) {
+                  args.add('-d')
+                }
+                if (ext.clean) {
+                  args.add('--rm')
+                } else {
+                  finalizedBy dockerRunStatus
+                }
                 for (String port : ext.ports) {
                     args.add('-p')
                     args.add(port)
                 }
+                for (Entry<String,String> volume : ext.volumes.entrySet()) {
+                    File localFile = new File(project.projectDir, volume.key)
+
+                    if (!localFile.exists()) {
```

Comment:
Ya, I went back and forth on that one, since it wasn't strictly an error like the port check you had in the extension. If you think it's reasonable to move there, I will.